### PR TITLE
fix #17758, deprecate `is`

### LINKED ISSUE #17758: make `===` primary, deprecate `is`
Since we now almost universally use the `is` function by the operator `===`, we should make that the official name and make `is` a deprecated alias for `===` and eventually free up the name `is`.

### PR DESCRIPTION
**Pull request by JeffBezanson**

### CODE REVIEW

File: base/associative.jl [hunk 1]
```diff
@@ -324,7 +323,7 @@ end
 
 function pop!(t::ObjectIdDict, key::ANY)
     val = pop!(t, key, secret_table_token)
-    !is(val,secret_table_token) ? val : throw(KeyError(key))
+    val !== secret_table_token ? val : throw(KeyError(key))
```
Comment (vtjnash) [inline base/associative.jl]: maybe check that this doesn't cause a performance regression?

File: base/inference.jl [hunk 2]
```diff
@@ -1403,8 +1403,8 @@ function smerge(sa::Union{NotFound,VarState}, sb::Union{NotFound,VarState})
     VarState(tmerge(sa.typ, sb.typ), sa.undef | sb.undef)
 end
 
-tchanged(n::ANY, o::ANY) = is(o,NF) || (!is(n,NF) && !(n ⊑ o))
-schanged(n::ANY, o::ANY) = is(o,NF) || (!is(n,NF) && !issubstate(n, o))
+tchanged(n::ANY, o::ANY) = o===NF || (n!==NF && !(n ⊑ o))
+schanged(n::ANY, o::ANY) = o===NF || (n!==NF && !issubstate(n, o))
```
Comment (vtjnash) [inline base/inference.jl]: also performance here (and so on)

File: test/sets.jl [hunk 3]
```diff
@@ -3,12 +3,12 @@
 # Set tests
 
 # Construction, collect
-@test is(typeof(Set([1,2,3])), Set{Int})
-@test is(typeof(Set{Int}([3])), Set{Int})
+@test ===(typeof(Set([1,2,3])), Set{Int})
+@test ===(typeof(Set{Int}([3])), Set{Int})
```
Comment (tkelman) [inline test/sets.jl]: these should probably be `isa` ?

Comment (JeffBezanson) [inline test/sets.jl]: Yes; this way was just easier search-replace.

File: base/associative.jl (hunk 1 above)
Comment (JeffBezanson) [inline base/associative.jl]: From what I can tell `!==` always gets inlined.

**Comment by JeffBezanson** [comment]
@nanosoldier `runbenchmarks(ALL, vs = ":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/18978#issuecomment-254340154) has completed - no performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/3a97fef02f1c41fb8a26f48f527e1e6f8c597d18/a6e0204_vs_1ea0de1/report.md). cc @jrevels

File: base/inference.jl (hunk 2 above)
Comment (TotalVerb) [inline base/inference.jl]: Inference performance regressions probably weren't caught by nanosoldier. I'd be surprised if this didn't regress, but it's probably not a big deal.

**Pull request merged**